frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -0,0 +1,82 @@
+import { BackingStorageType } from '../../constants/create-storage-system';
+import { ExternalProviderId } from '../../odf-external-providers/external-providers';
+
+export type WizardState = React.ReducerState<WizardReducer>;
+export type WizardDispatch = React.Dispatch<React.ReducerAction<WizardReducer>>;
+
+export type WizardCommonProps = {
+  state: WizardState;
+  dispatch: WizardDispatch;
+};
+
+/* State of CreateStorageSystem */
+export const initialState: CreateStorageSystemState = {
+  currentStep: 1,
+  backingStorage: {
+    type: BackingStorageType.EXISTING,
+    externalProvider: ExternalProviderId.RHCS,
+  },
+  storageClass: { name: '', provisioner: '' },
+};
+
+type CreateStorageSystemState = {
+  currentStep: number;
+  backingStorage: {
+    type: BackingStorageType;
+    externalProvider: ExternalProviderId;
+  };
+  storageClass: { name: string; provisioner?: string };
+};
+
+/* Reducer of CreateStorageSystem */
+export const reducer: WizardReducer = (prevState, action) => {
+  const newState = { ...prevState };
```

Comment:
```suggestion
  const newState = _.deepCopy(prevState);
```
You have nested properties, shallow copy is not sufficient.

Reply:
yeah! I failed to update wrt changes.